eq-author/src/components/PositionModal/index.js

```diff
@@ -32,64 +32,117 @@ const Trigger = styled.button.attrs({ type: "button" })`
   }
 `;
 
+const Indent = styled(Option)`
+  margin-left: ${({ indent }) => (indent ? 1 : 0)}em;
+`;
+
 const PositionModal = ({ options, onMove, selected }) => {
-  const previousPosition = useMemo(
-    () => options.findIndex(({ id }) => id === selected.id),
-    [options, selected]
-  );
+  const positionButtonId = uniqueId("PositionModal");
   const [isOpen, setIsOpen] = useState(false);
-  const [selectedPosition, setSelectedPosition] = useState(previousPosition);
-  const [positionButtonId] = useState(uniqueId("PositionModal"));
+  const previousIndex = options.findIndex(({ id }) => id === selected.id);
+  const previousPosition = useRef(previousIndex > -1 ? previousIndex : 0);
+  const [{ position, item }, setOption] = useState({
+    position: previousPosition.current,
+    item: options[previousPosition.current],
+  });
```

Comment:
I'm wondering if the variable names could be improved here. For instance:

`previousPosition` suggests it contains an integer, but actually it's the previous item in the list (I think). You could consider changing it to something like `previousItemRef`.

`previousIndex` becomes `previousItemIndex`.

```suggestion
  const previousItemIndex = options.findIndex(({ id }) => id === selected.id);
  const previousItemRef = useRef(previousItemIndex > -1 ? previousItemIndex : 0);
  const [{ position, item }, setOption] = useState({
    position: previousItemRef.current,
    item: options[previousItemRef.current],
  });
```

Reply:
previousPosition is an integer, `findIndex` returns a number